lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -634,11 +634,7 @@ public function getColumns()
         }
         $colNames = array_unique(array_merge($pkCols, $fkCols, array_keys($columns)));
 
-        uksort($columns, function ($a, $b) use ($colNames) {
-            return (array_search($a, $colNames) >= array_search($b, $colNames));
-        });
-
-        return $columns;
+        return array_replace(array_flip($colNames), $columns);
```

Comment:
That is indeed the "standard approach" when writing DDL, as it otherwise is really hard to figure out what is going on. Still, is there no `O(n * log(n))` or similar solution that doesn't make the code unreadable? Original code was heavy due to the repeated `array_search()` calls, which make this `O(n * log(n) * n * log(n))`, so your approach is good, just not understandable at all.